apps/labs/posts/building-scipy-with-flang.md

```diff
@@ -0,0 +1,598 @@
+---
+title: >
+  The 'eu' in eucatastrophe -- Why SciPy builds for Python 3.12 on Windows are a minor miracle
+published: November 5, 2023
+authors: [axel-obermeier]
+description: 'Moving SciPy to Meson meant finding a different Fortran compiler
+on Windows, which was particularly tricky to pull off for conda-forge. This
+blog tells the story about how things looked pretty grim for the Python 3.12
+release, and how things ended up working out just in the nick of time.'
+category: [Developer workflows]
+featuredImage:
+  src: /posts/building-scipy-with-flang/scipy_indy.jpg
+  alt: >
+    An object-labelling meme showing Indiana Jones (labelled "SciPy") running away in a tight space from a rolling boulder labelled "distutils removal", while carrying something labelled "Fortran code".
+hero:
+  imageSrc: /posts/building-scipy-with-flang/hero.png
+  imageAlt: 'A collage of the logos of SciPy, Meson, Fortran & LLVM. Some graphical elements on a white background surround the illustration.'
+---
+
+You've probably heard already that Python 3.12 was released recently.
+For a while already, we've been getting new feature releases every year,
+so perhaps this wasn't big news to you (though there's lots of interesting
+stuff in there!).
+
+It would appear even more "ordinary" that key packages (think `pandas`,
+`matplotlib` etc.) in the ecosystem would have a release compatible with the
+new Python version pretty soon after, and broadly speaking, you would be right.
+Not that there isn't a huge amount of dedicated maintainers (mostly
+volunteers!) working tirelessly behind the scenes to make that happen,
+but overall, it's a pretty routine situation.
+
+However, behind all the ordinary-seeming "SciPy released builds compatible with
+Python 3.12" hides an extraordinary story worth telling, because of how several
+unrelated, multi-decade-long timelines happened to clash in a way that could
+have very easily led to no Python 3.12-compatible releases for a long time.
+
+To understand why this was such a lucky coincidence (though we tried our best
+to tip the scales, _a lot_ of luck was necessary), we need to zoom out a bit
+and explore the big players involved in this situation.
+In the interest of brevity, the following recap is going to be incomplete and
+opinionated.
+From our perch on the shoulders of giants, let's take a quick look at the
+landscape.
+
+We'll briefly shed some light on the following:
+
+- Why is Fortran still used in so many places?
+- How is that relevant to Python?
+- Why progress in Python packaging standardisation throws a wrench in the works.
+- What role conda-forge plays in this context.
+
+If you feel you have a good-enough understanding of all this, or not much time,
+[skip](#The-situation) right past the background. 😉
```

Comment:
This link does not work in the rendered page; however, to the best of my understanding, it does follow the required pattern? Or do we need to also name the text itself according to the anchor?

Reply:
The auto-slugification of these fragments/anchors normalizes to all lowercase kebab-case (among other things), so we need the lowercase `t` here.

```suggestion
If you feel you have a good-enough understanding of all this, or not much time,
[skip](#the-situation) right past the background. 😉
```